run_dacapo.py

```diff
@@ -0,0 +1,141 @@
+JAVA_VMS["graal-ce-hotspot"] = [os.path.join(PATHS["GRAALCE_DIR"],
+                                             "bin", "java")]
+JAVA_VMS["graal-ce"] = [os.path.join(PATHS["GRAALCE_DIR"], "bin", "java"),
+                        "-XX:-EnableJVMCI", "-XX:-UseJVMCICompiler"]
+JAVA_VMS["openj9"] = [os.path.join(PATHS["OPENJ9_DIR"], "bin", "java")]
+
+
+def main():
+    for jvm_name, jvm_args in JAVA_VMS.items():
+        csvp = "dacapo.%s.results" % jvm_name
+        with open(csvp, 'w') as csvf:
+            writer = csv.writer(csvf)
+            writer.writerow(
+                ['processnum', 'benchmark'] + list(range(ITERATIONS)))
+
+            for benchmark in BENCHMARKS:
+                if benchmark in BROKEN[jvm_name]:
+                    print("!!! %s::%s -- Skipping all. Marked broken." %
+                          (jvm_name, benchmark))
+                    sys.stdout.flush()
+                    continue
+
+                for pexec_idx in range(PROCESSES):
+                    print(">>> %s::%s::%s" %
+                          (jvm_name, benchmark, str(pexec_idx)))
+                    sys.stdout.flush()
+
+                    # Flush the CSV writing, and then give the OS some time to
+                    # write stuff out to disk before running the next process
+                    # execution.
+                    csvf.flush()
+                    os.fsync(csvf.fileno())
+
+                    args = jvm_args + ["-Xms12G", "-Xmx12G", "-jar",
+                                       PATHS["DACAPO_JAR"], benchmark, "-n",
+                                       str(ITERATIONS + 1)]
+                    p = Popen(args, stderr=PIPE, stdout=PIPE)
+                    stdout, stderr = p.communicate()
+                    stdout, stderr = stdout.decode(), stderr.decode()
+
+                    if p.returncode != 0:
+                        sys.stderr.write(
+                            "\nWARNING: process execution crashed\n")
+                        sys.stderr.write("stdout:\n")
+                        sys.stderr.write(stdout + "\n")
+                        sys.stderr.write("\nstderr:\n")
+                        sys.stderr.write(stderr + "\n")
+                        sys.stderr.flush()
+                        writer.writerow([pexec_idx, benchmark, "crash"])
+                        continue
+
+                    # Dacapo offers no way to write the results to a file, so
+                    # we have the scrape the timings from stderr.
+                    output = []
+                    for line in stderr.splitlines():
+                        if not line.startswith("====="):
+                            continue
+                        if "completed warmup" not in line:
+                            continue
+                        assert benchmark in line
+                        line = line.split()
+                        index = line.index("in")
+                        assert line[index + 2] == "msec"
```

Comment:
Note that Dacapo now measures in milliseconds, whereas before it used nanoseconds. Please check the conversion to seconds below is correct.

Reply:
What units do we intend converting *into*?

Reply:
Seconds, as before.

Reply:
Then dividing by 1000 is fine :)